ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift

```diff
@@ -75,7 +76,32 @@ public class APIKeyboardRepository: KeyboardRepository {
     }
 
     options = result.options
-    languages = Dictionary(uniqueKeysWithValues: result.languages.map { ($0.id, $0) })
+
+    // Detect duplicate ids - this scenario indicates an infrastructure problem.
+    var dictionary = Dictionary<String, Language>()
```

Comment:
Wow. I think I preferred the earlier code for readability! Could you refactor this into a function?